### PR TITLE
Add arithmetic test cases for EVM->Ewasm translator

### LINKED ISSUE #7956: Add tests for the EVM to Ewasm Yul translator
- [ ] arithmetic (#8014)
- [x] bitwise (#7963)
- [x] comparison (#7977)
- [x] create (#7942) 
- [x] call* (#7942)

### PR DESCRIPTION
**Pull request by rodiazet**
Closes #7956.
Depends on #8015.

### CODE REVIEW

File: test/libyul/ewasmTranslationTests/arithmetic_addmod.yul [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  sstore(0, addmod(0, 1, 1))
```
Comment (axic) [inline test/libyul/ewasmTranslationTests/arithmetic_addmod.yul:2]:
```
std::exception::what: :2:13: Error: Function expects 8 arguments but got 12.
  sstore(0, addmod(0, 1, 1))
            ^----^
```

Need to fix the definition in the polyfill.

File: test/libyul/ewasmTranslationTests/arithmetic_mulmod.yul [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  sstore(0, mulmod(0, 1, 2))
```
Comment (axic) [inline test/libyul/ewasmTranslationTests/arithmetic_mulmod.yul:2]: Same here:
```
std::exception::what: :2:13: Error: Function expects 8 arguments but got 12.
  sstore(0, mulmod(0, 1, 2))
            ^----^
```

File: test/libyul/ewasmTranslationTests/arithmetic_sdiv.yul [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  sstore(0, sdiv(0, 1))
```
Comment (axic) [inline test/libyul/ewasmTranslationTests/arithmetic_sdiv.yul:2]: This is missing in the polyfill:
```
std::exception::what: :2:13: Error: Function not found.
  sstore(0, sdiv(0, 1))
            ^--^
```

Please add a dummy one with `unreachable`.

**Review by axic — CHANGES_REQUESTED** [review]
Please refill these test expectations with `develop` and not with #7766.

File: test/libyul/ewasmTranslationTests/arithmetic_sdiv.yul (hunk 3 above)
Comment (rodiazet) [reply test/libyul/ewasmTranslationTests/arithmetic_sdiv.yul:2]: Done

File: test/libyul/ewasmTranslationTests/arithmetic_mulmod.yul (hunk 2 above)
Comment (rodiazet) [reply test/libyul/ewasmTranslationTests/arithmetic_mulmod.yul:2]: done

File: test/libyul/ewasmTranslationTests/arithmetic_addmod.yul (hunk 1 above)
Comment (rodiazet) [reply test/libyul/ewasmTranslationTests/arithmetic_addmod.yul:2]: done

**Comment by axic** [comment]
@rodiazet can you squash these commits?

**Comment by rodiazet** [comment]
> @rodiazet can you squash these commits?

Done

**Comment by axic** [comment]
I've started reviewing this, but seems easier after merging #8015, because then we know which ones supposed to work and which don't.

**Comment by chriseth** [comment]
Merged #8015

**Comment by axic** [comment]
@rodiazet can you rebase please?

File: libyul/backends/wasm/EVMToEWasmTranslator.cpp [hunk 4]
```diff
@@ -146,6 +146,10 @@ function smod(x1, x2, x3, x4, y1, y2, y3, y4) -> r1, r2, r3, r4 {
 	// TODO implement properly
 	r4 := i64.rem_u(x4, y4)
 }
+function sdiv(x1, x2, x3, x4, y1, y2, y3, y4) -> r1, r2, r3, r4 {
```
Comment (axic) [inline libyul/backends/wasm/EVMToEWasmTranslator.cpp]: Can you move this after `dev`?

File: test/libyul/EWasmTranslationTest.cpp [hunk 5]
```diff
@@ -131,7 +131,7 @@ string EWasmTranslationTest::interpret()
 {
 	InterpreterState state;
 	state.maxTraceSize = 10000;
-	state.maxSteps = 10000;
+	state.maxSteps = 100000;
```
Comment (axic) [inline test/libyul/EWasmTranslationTest.cpp:134]: Is this still needed?

File: libyul/backends/wasm/EVMToEWasmTranslator.cpp (hunk 4 above)
Comment (rodiazet) [reply libyul/backends/wasm/EVMToEWasmTranslator.cpp]: OK.

File: test/libyul/EWasmTranslationTest.cpp (hunk 5 above)
Comment (rodiazet) [reply test/libyul/EWasmTranslationTest.cpp:134]: Yes. For mod, addmod and mulmod test cases. These test cases need more then 1000 steps to run.

**Comment by rodiazet** [comment]
> @rodiazet can you rebase please?

Done

**Comment by axic** [comment]
Sorry I think you need to refill this also (please make sure to squash all of that).

**Review by axic — DISMISSED** [review]

**Comment by axic** [comment]
Sorry to bug you again, but can rebase (or enable edits from maintainers)? I think tests will work again after a rebase.

**Comment by rodiazet** [comment]
> Sorry to bug you again, but can rebase (or enable edits from maintainers)? I think tests will work again after a rebase.

Nothing to be sorry about. Rebased.

**Review by axic — APPROVED** [review]

**Pull request merged**